#! /bin/sh /usr/share/dpatch/dpatch-run
## 29-bug2262-fix.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Backport bug#2262 fix.

@DPATCH@

--- ACE_wrappers/TAO/orbsvcs/orbsvcs/Notify/Service.h-
+++ ACE_wrappers/TAO/orbsvcs/orbsvcs/Notify/Service.h
@@ -21,9 +21,25 @@
 
 #include "ace/Service_Object.h"
 #include "ace/Service_Config.h"
-#include "ace/Service_Object.h"
-#include "orbsvcs/CosNotifyChannelAdminS.h"
-#include "tao/PortableServer/PortableServer.h"
+#include "ace/CORBA_macros.h"
+
+namespace PortableServer
+{
+  class POA;
+  typedef POA *POA_ptr;
+}
+
+namespace CosNotifyChannelAdmin
+{
+  class EventChannelFactory;
+  typedef EventChannelFactory *EventChannelFactory_ptr;
+}
+
+namespace CORBA
+{
+  class ORB;
+  typedef ORB *ORB_ptr;
+}
 
 /**
  * @class TAO_Notify_Service
@@ -36,6 +52,9 @@
 class TAO_Notify_Serv_Export TAO_Notify_Service : public ACE_Service_Object
 {
 public:
+  /// Destructor.
+  virtual ~TAO_Notify_Service (void);
+
   /// Init the service from configurator
   virtual int init (int argc, ACE_TCHAR *argv[]) = 0;
 
--- ACE_wrappers/TAO/orbsvcs/orbsvcs/Notify/Service.cpp-
+++ ACE_wrappers/TAO/orbsvcs/orbsvcs/Notify/Service.cpp
@@ -0,0 +1,12 @@
+// Service.cpp,v 1.1 2005/10/13 11:53:17 jwillemsen Exp
+
+#include "Service.h"
+
+ACE_RCSID (Notify,
+           TAO_Notify_Service,
+           "Service.cpp,v 1.1 2005/10/13 11:53:17 jwillemsen Exp")
+
+TAO_Notify_Service::~TAO_Notify_Service ()
+{
+}
+
--- ACE_wrappers/TAO/orbsvcs/orbsvcs/CosNotification.mpc-
+++ ACE_wrappers/TAO/orbsvcs/orbsvcs/CosNotification.mpc
@@ -200,6 +200,7 @@
       Notify/Routing_Slip.cpp
       Notify/Routing_Slip_Persistence_Manager.cpp
       Notify/Routing_Slip_Queue.cpp
+      Notify/Service.cpp
       Notify/Subscription_Change_Worker.cpp
       Notify/Supplier.cpp
       Notify/SupplierAdmin.cpp
